File: apps/web/src/types/work-order.ts
export type WorkOrderStatus = "PENDIENTE" | "EN_PROGRESO" | "COMPLETADA" | "CANCELADA";
export type WorkOrderItemType = "LABOR" | "PARTS";

export interface WorkOrderItem {
  id: string;
  description: string;
  type: WorkOrderItemType;
  quantity: number;
  unit_price: number;
  total: number;
}

export interface WorkOrder {
  id: string;
  reception_id: string;
  vehicle_label: string;
  customer_id: string;
  customer_name: string;
  status: WorkOrderStatus;
  diagnosis: string | null;
  assigned_to: string | null;
  estimated_cost: number | null;
  actual_cost: number | null;
  items: WorkOrderItem[];
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

export interface WorkOrdersPage {
  data: WorkOrder[];
  meta: { total: number; page: number; limit: number };
}

export interface CreateWorkOrderRequest {
  receptionId: string;
  diagnosis?: string;
  assignedTo?: string;
  estimatedCost?: number;
}

export interface AddWorkOrderItemRequest {
  description: string;
  type: WorkOrderItemType;
  quantity: number;
  unitPrice: number;
}

File: apps/api/src/main/java/rd/tallerfacil/api/workorder/domain/WorkOrder.java
package rd.tallerfacil.api.workorder.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rd.tallerfacil.api.reception.domain.Reception;
import rd.tallerfacil.api.shared.domain.BaseEntity;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "work_orders")
@Getter
@Setter
@NoArgsConstructor
public class WorkOrder extends BaseEntity {

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reception_id", nullable = false, unique = true)
    private Reception reception;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private WorkOrderStatus status = WorkOrderStatus.PENDIENTE;

    @Column(columnDefinition = "TEXT")
    private String diagnosis;

    @Column(name = "assigned_to", length = 100)
    private String assignedTo;

    @Column(name = "estimated_cost", precision = 10, scale = 2)
    private BigDecimal estimatedCost;

    @Column(name = "actual_cost", precision = 10, scale = 2)
    private BigDecimal actualCost;

    @Column(name = "started_at")
    private Instant startedAt;

    @Column(name = "completed_at")
    private Instant completedAt;

    @OneToMany(mappedBy = "workOrder", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("createdAt ASC")
    private List<WorkOrderItem> items = new ArrayList<>();

    public void transitionTo(WorkOrderStatus next) {
        if (!this.status.canTransitionTo(next)) {
            throw new IllegalStateException(
                    "No se puede cambiar estado de " + this.status + " a " + next);
        }
        if (next == WorkOrderStatus.EN_PROGRESO) {
            this.startedAt = Instant.now();
        }
        if (next == WorkOrderStatus.COMPLETADA) {
            this.completedAt = Instant.now();
            this.actualCost = items.stream()
                    .map(WorkOrderItem::getTotal)
                    .reduce(BigDecimal.ZERO, BigDecimal::add);
        }
        this.status = next;
    }
}
